### ISSUE TITLE
Apollo v3 subscriptions not working on React Native Android only.

### ISSUE THREAD

**Issue by immortalx** [apollographql/apollo-client#6520]
Just trying to upgrade to v3. 

Everything is working fine, just subscriptions on React Native android doesn't work at all. On iOS is working fine as expected.
On v2, both iOS and Android works fine.

**Intended outcome:**
Subscriptions should work on both iOS and Android while using React Native.

**Actual outcome:**
Subscriptions only works in iOS. On android nothing happens, no errors or warnings.

**Versions**
  npmPackages:
    @apollo/client: ^3.0.0-rc.9 => 3.0.0-rc.9 
    @apollo/link-context: ^2.0.0-beta.3 => 2.0.0-beta.3 
    @apollo/link-error: ^2.0.0-beta.3 => 2.0.0-beta.3 
    @apollo/link-retry: ^2.0.0-beta.3 => 2.0.0-beta.3 
    @apollo/link-ws: ^2.0.0-beta.3 => 2.0.0-beta.3 
    apollo-cache-persist: ^0.1.1 => 0.1.1

**Comment by mschipperheyn** [comment]
I am testing on Android right now. It does work for me.

**Comment by martimalek** [comment]
I'm also experiencing this issue. I'm using the basic setup from the docs:
```
const httpLink = createHttpLink({ uri: `${API_URL}/graphql` });

const wsLink = new WebSocketLink({
    uri: `${SUBSCRIPTION_API_URL}/graphql`,
    options: {
        reconnect: true,
    },
});

const splitLink = split(
    ({ query }) => {
        const definition = getMainDefinition(query);
        return (
            definition.kind === 'OperationDefinition'
            && definition.operation === 'subscription'
        );
    },
    wsLink,
    httpLink,
);
```
the split is working as expected but when using `useSubscription` everything in the response is undefined except the `loading` which is always true. On **GraphiQL** and on **React** (non-native) it is working perfectly.

**Comment by benjamn** [comment]
Can you try different beta/RC versions (`beta.1`-`beta.54`, `rc.0`-`rc.10`) to determine if this was a regression introduced in a specific release? A binary search will probably save time, but I would check `beta.1` first to see if it was working, before you do a full search.

**Comment by immortalx** [comment]
@benjamn You're right,it's a regression! Just tried a few beta and RC versions, on 3.0.0-beta.23 it works as expected.

My setup:

```javascript
let persistor
export const wsClient = new SubscriptionClient(WS_URI, {
  lazy: true,
  reconnect: true,
  connectionParams: async () => {
    const token = await AsyncStorage.getItem(authToken)
    return {
      Authorization: token ? `Bearer ${token}` : null
    }
  }
})
const wsLink = new WebSocketLink(wsClient)
export const setupApollo = async () => {
  const retryLink = new RetryLink({
    attempts: (count, operation, error) => {
      const isMutation =
        operation &&
        operation.query &&
        operation.query.definitions &&
        Array.isArray(operation.query.definitions) &&
        operation.query.definitions.some(
          (def) =>
            def.kind === 'OperationDefinition' && def.operation === 'mutation'
        )
      if (isMutation) {
        return !!error && count < 25
      }
      return !!error && count < 6
    }
  })
  const httpLink = new HttpLink({
    uri: GRAPHQL_URI,
    credentials: 'include'
  })

  const splitLink = split(
    ({ query }) => {
      const definition = getMainDefinition(query)
      return (
        definition.kind === 'OperationDefinition' &&
        definition.operation === 'subscription'
      )
    },
    wsLink,
    httpLink
  )

  const errorLink = onError(({ graphQLErrors, networkError }) => {
    if (graphQLErrors)
      graphQLErrors.map(({ message, locations, path }) => {
        console.log(
          `[GraphQL error]: Message: ${message}, Location: ${locations}, Path: ${path}`
        )
      })
    if (networkError) console.log(`[Network error]: ${networkError}`)
  })

  const authLink = setContext(async (_, { headers }) => {
    const token = await AsyncStorage.getItem(authToken)
    return {
      headers: {
        ...headers,
        authorization: token ? `Bearer ${token}` : null
      }
    }
  })
  const cache = new InMemoryCache({
    possibleTypes: introspectionResult.possibleTypes
  })
  persistor = new CachePersistor({
    cache,
    storage: AsyncStorage as PersistentStorage<
      PersistedData<NormalizedCacheObject>
    >,
    maxSize: 2048000,
    // debug: true,
    trigger: 'background'
  })

  const currentVersion = await AsyncStorage.getItem(SCHEMA_VERSION_KEY)
  try {
    if (currentVersion === SCHEMA_VERSION) {
      // If the current version matches the latest version,
      // we're good to go and can restore the cache.
      await persistor.restore()
    } else {
      // Otherwise, we'll want to purge the outdated persisted cache
      // and mark ourselves as having updated to the latest version.
      await persistor.purge()
      await AsyncStorage.setItem(SCHEMA_VERSION_KEY, SCHEMA_VERSION)
    }
  } catch (err) {
    // CachePersistor failed - purge
    await persistor.purge()
  }

  return new ApolloClient({
    link: from([errorLink, authLink, retryLink, splitLink]),
    cache,
    assumeImmutableResults: true,
    queryDeduplication: true,
    defaultOptions: {
      watchQuery: {
        fetchPolicy: 'cache-and-network',
        errorPolicy: 'none'
      },
      query: {
        fetchPolicy: 'cache-first',
        errorPolicy: 'all'
      }
    }
  })
}

```

**Comment by immortalx** [comment]
@benjamn Lastest working beta is beta.45, i hope it helps.

**Comment by martimalek** [comment]
@immortalx is correct, **beta.46** is the first one where it breaks

**Comment by benjamn** [comment]
@immortalx Any chance you could put your code into a runnable (React Native) reproduction repo/sandbox? Note that AC3 is slated to be released tomorrow, so this might not get fixed before then, but that doesn't mean we won't fix it.

**Comment by immortalx** [comment]
@benjamn sure, can you point me to any public graphql api with subscriptions so i can use it to test?

**Comment by mikeislearning** [comment]
Thanks for bringing this up @immortalx ,I was starting to go nuts. I can confirm that this is still broken in the latest 3.1.1 version, and I'm now using @apollo/client^3.0.0-beta.45 to get subscriptions working on android

**Comment by schonfeld** [comment]
Same in v3.1.3. Same Subscriptions code that worked fine in v2 on both {iOS, Android} is now broken on Android. The `loading` flag is just stuck at `true`.

**Comment by schonfeld** [comment]
Worth noting that when rn's Debug mode is enabled, causing JS to be parsed in the browser context (as opposed to the "device" or bridge), subscriptions work just fine.

**Comment by schonfeld** [comment]
Some useful insight: turning off Hermes engine actually fixed the issue. This definitely smells like a js engine change/issue, rather than Apollo client problem.

**Comment by kirantripathi** [comment]
<img width="642" alt="Screen Shot 2020-08-29 at 19 59 05" src="https://user-images.githubusercontent.com/17666887/91638944-c570f480-ea32-11ea-911e-e0b86892fc4b.png">
<img width="584" alt="Screen Shot 2020-08-29 at 19 58 52" src="https://user-images.githubusercontent.com/17666887/91638950-d6ba0100-ea32-11ea-88ea-327d94967eb9.png">
<img width="579" alt="Screen Shot 2020-08-29 at 19 58 27" src="https://user-images.githubusercontent.com/17666887/91638956-e20d2c80-ea32-11ea-85c4-6bb71e755002.png">

My subscription hook is just returning loading as response. I am using  "subscriptions-transport-ws": "^0.9.18". Do you guys have any idea about how to fix it? @schonfeld  @benjamn @immortalx @martimalek

**Comment by schonfeld** [comment]
> <img alt="Screen Shot 2020-08-29 at 19 59 05" width="642" src="https://user-images.githubusercontent.com/17666887/91638944-c570f480-ea32-11ea-911e-e0b86892fc4b.png">
> 
> <img alt="Screen Shot 2020-08-29 at 19 58 52" width="584" src="https://user-images.githubusercontent.com/17666887/91638950-d6ba0100-ea32-11ea-88ea-327d94967eb9.png">
> 
> <img alt="Screen Shot 2020-08-29 at 19 58 27" width="579" src="https://user-images.githubusercontent.com/17666887/91638956-e20d2c80-ea32-11ea-85c4-6bb71e755002.png">
> 
> My subscription hook is just returning loading as response. I am using "subscriptions-transport-ws": "^0.9.18". Do you guys have any idea about how to fix it? @schonfeld @benjamn @immortalx @martimalek

If this only happens on Android for you, my solution was turning off Herms engine in `android/app/build.gradle`:
```gradle
project.ext.react = [
    entryFile: "index.js",
    enableHermes: false
]
```

If the issue also occurs on an iPhone sim (or device), your problem might not relate to this specific issue...

**Comment by kirantripathi** [comment]
> > <img alt="Screen Shot 2020-08-29 at 19 59 05" width="642" src="https://user-images.githubusercontent.com/17666887/91638944-c570f480-ea32-11ea-911e-e0b86892fc4b.png">
> > <img alt="Screen Shot 2020-08-29 at 19 58 52" width="584" src="https://user-images.githubusercontent.com/17666887/91638950-d6ba0100-ea32-11ea-88ea-327d94967eb9.png">
> > <img alt="Screen Shot 2020-08-29 at 19 58 27" width="579" src="https://user-images.githubusercontent.com/17666887/91638956-e20d2c80-ea32-11ea-85c4-6bb71e755002.png">
> > My subscription hook is just returning loading as response. I am using "subscriptions-transport-ws": "^0.9.18". Do you guys have any idea about how to fix it? @schonfeld @benjamn @immortalx @martimalek
> 
> If this only happens on Android for you, my solution was turning off Herms engine in `android/app/build.gradle`:
> 
> ```groovy-gradle
> project.ext.react = [
>     entryFile: "index.js",
>     enableHermes: false
> ]
> ```
> 
> If the issue also occurs on an iPhone sim (or device), your problem might not relate to this specific issue...

No, it is happening in both ios and android

**Comment by happyfloat** [comment]
Having the same problem with hermes. It's only working in debug mode because of the different js engine... damn...

**Comment by stackia** [comment]
Any updates? Subscription is still broken on Hermes engine.

**Comment by Rieranthony** [comment]
Any news at all? This issue is blocking our Android app, would love to start using v3 but this is a huge blocker :( Thanks!

**Comment by ryoid** [comment]
Also experiencing this issue, works on iOS. Using hermes engine for android.

    "react-native": "0.63.2",
    "@apollo/client": "^3.2.0",
    "subscriptions-transport-ws": "^0.9.18"

**Comment by rarenatoe** [comment]
I'd happily downgrande to @apollo/client@3.0.0-beta.45 but it seems that this version does not have `@apollo/client/link/context` yet which my app uses. So I am effectively screwed.

I already have hermes disabled by default.

Any update on this?

**Comment by rarenatoe** [comment]
I managed to get it back working. The error seems to have been that one of my susbscriptions didn't have an `OperationName`, i.e. it was written as:
```
export const mySubscription = gql`
  subscription {
    subscriptionName {
      id
      ...etc
    }
  }
`
```
This was fixed by giving it an `operationName` such as:
```
export const mySubscription = gql`
  subscription mySubscription {
    subscriptionName {
      id
      ...etc
    }
  }
`
```

In my case it just so happened that this subscription was the first one ever called and so for whatever reason it broke everything subscription-related after this even when other subscriptions did have an `OperationName`.

For whatever reason Apollo 3 now requires all subscriptions to have an OperationName, which definitely was never the case in 2.X. This was never mentioned anywhere, and let's just say I am upset.

**Comment by therealemjy** [comment]
The error persists for me on Apollo v3, even when giving the subscription an `OperationName`:
```graphql
const MESSAGE_ADDED_SUBSCRIPTION = gql`
  subscription messageAdded($chatIds: [ID]!) {
    messageAdded(chatIds: $chatIds) {
      _id
      createdAt
      senderId
      chatId
      content
    }
  }
`;
```

The issue happens only on Android when Hermes engine is enabled.

**Comment by dvirben123** [comment]
> The error persists for me on Apollo v3, even when giving the subscription an `OperationName`:
> 
> ```graphql
> const MESSAGE_ADDED_SUBSCRIPTION = gql`
>   subscription messageAdded($chatIds: [ID]!) {
>     messageAdded(chatIds: $chatIds) {
>       _id
>       createdAt
>       senderId
>       chatId
>       content
>     }
>   }
> `;
> ```
> 
> The issue happens only on Android when Hermes engine is enabled.

same thing here, Apollo v3, can't get subscription working (thinking moving to socket.io or pusher)

**Comment by zakthedev** [comment]
Hi @dvirben123, This one is not related to the issue but can I use Subscriptions for push notifications on mobile? is the app going to still receive data after subscribing when the app is in the background or killed? and what strategy is used to push pending notifications after device is turned on?

You may or may not have the answers to all of these but I really want to understand the concept here so any clarifications would be appreciated. Thank you :)

**Comment by dvirben123** [comment]
> Hi @dvirben123, This one is not related to the issue but can I use Subscriptions for push notifications on mobile? is the app going to still receive data after subscribing when the app is in the background or killed? and what strategy is used to push pending notifications after device is turned on?
> 
> You may or may not have the answers to all of these but I really want to understand the concept here so any clarifications would be appreciated. Thank you :)

Hi, basically I want to get notifications only when the app is live or in the background, I know that using Apple or Google services for push notification is a better approach, I want to know what is the maximum that I get can from Apollo

**Comment by kirantripathi** [comment]
subscription is inconsistent sometimes data is present and sometimes it is not returning data , and when in debug mode sometimes subscription doesn't return data. thinking of moving back to "apollo 3.0.0-beta.45" , will it solve this inconsistent issue. I have operation name too in my subscription. @zakthedev  @dvirben123  @rarenatoe  @Rieranthony  @immortalx  @benjamn  .any idea on how to solve this, or how did you guys fix this?

**Comment by kirantripathi** [comment]
subscription is inconsistent sometimes data is present and sometimes it is not returning data , and when in debug mode sometimes subscription doesn't return data. thinking of moving back to "apollo 3.0.0-beta.45" , will it solve this inconsistent issue. I have operation name too in my subscription. @zakthedev  @dvirben123  @rarenatoe  @Rieranthony  @immortalx  @benjamn  .any idea on how to solve this, or how did you guys fix this?

**Comment by martimalek** [comment]
Hi @kirantripathi! Our team has been stuck in **3.0.0-beta.45** for quite some time now (even months maybe?). AFAIK this specific version is where the least amount of problems happen.
Sadly we are currently having another subscription-related issue and are considering having a polling system at the same time as the subscriptions. It's a pity that subscriptions are so inconsistent but honestly we prefer patching them and continue using apollo rather than changing our entire system to something else, plus other than subscriptions we love **Apollo GraphQL** and use it everywhere we can.

**Comment by Nikosmonaut** [comment]
+1 does not work with hermes.

**Comment by kirantripathi** [comment]
<img width="720" alt="Screen Shot 2020-11-17 at 16 18 57" src="https://user-images.githubusercontent.com/17666887/99379402-bb071e00-28f0-11eb-813f-0a68cf4a3414.png">
<img width="576" alt="Screen Shot 2020-11-17 at 16 18 45" src="https://user-images.githubusercontent.com/17666887/99379417-bfcbd200-28f0-11eb-845a-ae55071985ff.png">

I don't have hermes enabled . this is my setup , I went back to using "apollo-link-ws" for web socket link 
and  "@apollo/client": "3.0.0-beta.45", but still the subscription is highly inconsistent.

**Comment by martimalek** [comment]
Hi @kirantripathi! Could you try importing `InMemoryCache` from **@apollo/client** instead of **apollo-boost**. I think that **apollo-boost** doesn't support subscriptions. Also I would recommend updating the dependencies to use the ones under the **@apollo** organization, so your imports should look kinda like this:
```
import { ApolloClient, InMemoryCache, createHttpLink, gql, ApolloLink, split } from '@apollo/client';
import { getMainDefinition } from '@apollo/client/utilities';
import { WebSocketLink } from '@apollo/link-ws';
import { onError } from '@apollo/link-error';
import { setContext } from '@apollo/link-context';
```

Good luck!

**Comment by Nikosmonaut** [comment]
I've found what is causing the problem in my case (not working with Hermes). But I'm not sure about what's really going on, and why there is a use of Symbol.species here. If someone better understands what is it for ?

Observable.js
```
function getSpecies(obj) {
  let ctor = obj.constructor;
  if (ctor !== undefined) {
    ctor = ctor[SymbolSpecies];
    if (ctor === null) {
      ctor = undefined;
    }
  }
  return ctor !== undefined ? ctor : Observable;
}

map(fn) {
  if (typeof fn !== 'function')
    throw new TypeError(fn + ' is not a function');

  let C = getSpecies(this);

  return new C(observer => this.subscribe({
    next(value) {
      try { value = fn(value) }
      catch (e) { return observer.error(e) }
      observer.next(value);
    },
    error(e) { observer.error(e) },
    complete() { observer.complete() },
  }));
 }
```

If we just replace C by Observable subscriptions work again

```
  return new Observer(observer => this.subscribe({
```

**Comment by kirantripathi** [comment]
If it is working with hermes disabled for you , can you share your setup file and apollo and react-native version as I am facing problem for subscription. @Nikosmonaut

**Comment by Nikosmonaut** [comment]
I always use the latest versions if it works.

package.json
```
    "@apollo/client": "^3.2.2",
    "@apollo/link-context": "^2.0.0-beta.3",
    "@apollo/link-error": "^2.0.0-beta.3",
    "@apollo/link-ws": "^2.0.0-beta.3",
    "apollo-cache-persist-dev": "^0.2.1",
    "apollo-upload-client": "^14.0.0"
    "graphql": "^15.0.0",
    "graphql-tag": "^2.10.3"
    "react": "16.13.1",
    "react-apollo": "^3.0.0",
    "react-native": "0.63.3"
    "subscriptions-transport-ws": "^0.9.15"
```

ApolloClient (basic)
```
import { Platform } from 'react-native';
import { ApolloLink, split, Observable } from '@apollo/client/core';
import { WebSocketLink } from '@apollo/link-ws';
import { getMainDefinition } from '@apollo/client/utilities';
import { onError } from '@apollo/link-error';
import { setContext } from '@apollo/link-context';
import { createUploadLink } from 'apollo-upload-client';
...
      _wsLink = new WebSocketLink({
        uri: AppConfig.server.socketUrl,
        options: {
          lazy: true,
          reconnect: true,
          connectionParams: () => ({
            authorization: _currentAuthorization,
            name: MYAPPNAME,
            version: AppConfig.version,
          }),
        },
      });

      mainLink = split(
        // split based on operation type
        ({ query }) => {
          const { kind, operation } = getMainDefinition(query);
          return kind === 'OperationDefinition' && operation === 'subscription';
        },
        _wsLink,
        httpLink,
      );
```

build.gradle
```
        buildToolsVersion = "29.0.3"
        minSdkVersion = 23
        compileSdkVersion = 29
        targetSdkVersion = 29
        supportLibVersion = "28.0.0"
        androidMapsUtilsVersion = "0.5+"
```

Not sure what else would be relevant to you ?

**Comment by Dunos** [comment]
@Nikosmonaut if you are using Apollo 3, you should probably drop all those link packages since they are now included in the main Apollo Client (check https://www.apollographql.com/docs/react/migrating/apollo-client-3-migration/ )

**Comment by kirantripathi** [comment]
@Dunos  is subscription working perfectly in your application, if it is can you share your apollo setup  and package imported so that I can replicate it and test it

**Comment by happyfloat** [comment]
@kirantripathi I created a gist for you with my working setup. It seems to be working very well and stable so far. (As long as I do not use hermes engine).

https://gist.github.com/happyfloat/c0ddfc933cfa6a40f71947ae0ff49767

**Comment by kirantripathi** [comment]
@Dunos  @Nikosmonaut  , one thing I am getting in subscription query response is regarding token expire , so if our token is expire , how can I reconnect the web socket with new token. any idea on that? 
I have added token expire logic for HTTP query but not sure on web socket.

**Comment by Dunos** [comment]
@kirantripathi the approach we use for authentication on websockets is to use a query parameter with the token in the WS uri, and refreshing the token and the uri using the "onError" events of the SubscriptionClient.

**Comment by kirantripathi** [comment]
@Dunos  can you provide me  example  ?

**Comment by Nikosmonaut** [comment]
@kirantripathi It's not really important to have a stale token in subscriptions. Because the connection is done only once the first time. Nevertheless if you want to reconnect you'll have to send a new token.

First of all check this library [subscriptions-transport-ws](https://github.com/apollographql/subscriptions-transport-ws) because it's behind all of it.

If you have followed the template from graphql docs, you would already have a base. Then use a function as connectionParams
```
      _wsLink = new WebSocketLink({
        uri: URI,
        options: {
          lazy: true,
          reconnect: true,
          connectionParams: () => ({
            authorization: YOURTOKEN,
          }),
        },
      });
```

Then when you get a expired token message replace your variable YOURTOKEN by the new one and reconnect your socket.
```
// This will close the current one and reconnect, calling your param function with new token
_wsLink.subscriptionClient.close(false);
```

**Comment by hannojg** [comment]
Tried many many different versions, jsut got it working on 3.0.0 beta 45, any version after that doesn't seem to work on hermes.

**Comment by kirantripathi** [comment]
@Nikosmonaut  I try without updating token , and it is working , so we won't need to update token after session expire. One thing is I am not able to find is the reason for inconsistency , however when I close the app and open again maybe it will reconnect again so that's why it start to work again, but sometimes even though I get information as reconnecting and reconnected the subscription don't work , and it only work after reloading the app. any idea on how can I tackle it .

**Comment by Nikosmonaut** [comment]
@kirantripathi I can't say for sure. But when you go background with your app the connection gets disconnected no matter what after some time (~1min). So when you go active again the connection is initiated once more. Imagine you have an expired token it will not work.

**Comment by benjamn** [comment]
re: https://github.com/apollographql/apollo-client/issues/6520#issuecomment-729038382

@Nikosmonaut You're looking at [code in the `zen-observable` library](https://github.com/zenparsing/zen-observable/blob/328fb66a99242900c0fd4a330e9ff66ecfa3a887/src/Observable.js#L29-L38) there.

`Observable` subclasses are supposed to define a static `Symbol.species` property so inherited `Observable` methods (like `Observable.prototype.map`) that need to create new `Observable` objects can use `new this.constructor[Symbol.species](observer => ...)` instead of just `new this.constructor(observer => ...)`, which is important specifically when the subclass constructor takes different/incompatible arguments than the `Observable` constructor takes. As a hint to methods like `map`, the subclass sets `Subclass[Symbol.species] = Observable` to prevent the subclass's own (incompatible) constructor from being invoked.

Apollo Client has one `Observable` subclass that needs `Symbol.species` [here](https://github.com/apollographql/apollo-client/blob/9ae90fe86dd41373179fd73d1b38552ebbc39602/src/utilities/observables/Concast.ts#L241-L253), and that code happens to have been introduced in `@apollo/client@3.0.0-beta.46`, so **I'm pretty sure you're onto something here!**

However, I'm having trouble guessing anything more about the problem, because vanilla [Hermes](https://github.com/facebook/hermes) purposefully [does not implement `Symbol.species`](https://github.com/facebook/hermes/blob/master/doc/Features.md#excluded-from-support), so I would expect this code to be skipped in Hermes, unless `Symbol.species` has been imperfectly polyfilled somehow? Not running is fine, because `getSpecies(concast)` just returns `Observable` in that case, which is what we wanted anyway.

I realize this was a complicated explanation that probably sounds borderline unhinged to programmers who aren't familiar with the ECMAScript specification, but hopefully it inspires someone here to dig a little deeper into whatever React Native is doing with `Symbol.species` that doesn't play nicely with Hermes.

**Comment by Nikosmonaut** [comment]
I had got to read it 3 times, but finally I got it.

Your suppositions are right. I think it's core-js that add @@species prop to Symbol => [core-js observable](https://github.com/zloirock/core-js#observable) which is indeed used by react-native => @react-native-community/cli => metro => metro-babel-register -> core-js

**Comment by benjamn** [comment]
Alright everybody, if you run `npm i @apollo/client@beta` (or `npm i @apollo/client@3.4.0-beta.1` specifically), you'll get a version that includes #7403, which may address some of the problems reported in this issue, such as https://github.com/apollographql/apollo-client/issues/6520#issuecomment-729038382. I don't expect this to solve all problems at once, but it should help focus the discussion somewhat. Thanks!

**Comment by benjamn** [comment]
Of course, if you're updating to v3.4 from v3.0.0-beta.45, you may want to read through the [`CHANGELOG.md`](https://github.com/apollographql/apollo-client/blob/main/CHANGELOG.md) to make sure you're on top of all the other changes. If that's totally infeasible, we could publish a one-off version based on `-beta.46` to minimize the immediate pain of updating, but I'm hoping everyone will eventually update to the latest version once this issue is fixed.

**Comment by vamshi9666** [comment]
@benjamn yes. this beta fixed my issue in android (hermes enabled) with version ```3.4.0-beta.3```

**Comment by martimalek** [comment]
From `3.4.0-beta.1` it works for us, thank you @benjamn!

**Comment by hannojg** [comment]
Same here 👋 works now

**Comment by benjamn** [comment]
The `Symbol.species` fix has been backported to `@apollo/client@3.3.8`, in case anyone was waiting for it (and unwilling/unable to use the v3.4 betas).

**Comment by quannguyennn** [comment]
For some reason, subcription not working for me =( loading state true, data and error are undefined. Im using apollo client 3.3.8. Im got data on playground however

**Comment by Necromant1k** [comment]
> For some reason, subcription not working for me =( loading state true, data and error are undefined. Im using apollo client 3.3.8. Im got data on playground however

Having the same issue 

```"react": "17.0.1",
    "react-native": "0.64.2",```
